crates/router/tests/connectors/payu.rs

```diff
@@ -0,0 +1,214 @@
+    connector_auth,
+    utils::{self, ConnectorActions},
+};
+
+struct Payu;
+impl ConnectorActions for Payu {}
+impl utils::Connector for Payu {
+    fn get_data(&self) -> types::api::ConnectorData {
+        use router::connector::Payu;
+        types::api::ConnectorData {
+            connector: Box::new(&Payu),
+            connector_name: types::Connector::Payu,
+            get_token: types::api::GetToken::Connector,
+        }
+    }
+
+    fn get_auth_token(&self) -> types::ConnectorAuthType {
+        types::ConnectorAuthType::from(
+            connector_auth::ConnectorAuthentication::new()
+                .payu
+                .expect("Missing connector authentication configuration"),
+        )
+    }
+
+    fn get_name(&self) -> String {
+        "payu".to_string()
+    }
+}
+
+#[actix_web::test]
+async fn should_authorize_card_payment() {
+    let authorize_response = Payu {}.authorize_payment(None).await;
+    assert_eq!(authorize_response.status, enums::AttemptStatus::Pending);
+    if let Some(transaction_id) = utils::get_connector_transaction_id(authorize_response) {
+        let sync_response = Payu {}
+            .sync_payment(Some(types::PaymentsSyncData {
+                connector_transaction_id: router::types::ResponseId::ConnectorTransactionId(
+                    transaction_id.clone(),
+                ),
+                encoded_data: None,
+            }))
+            .await;
+        assert_eq!(sync_response.status, enums::AttemptStatus::Authorized);
+    }
+}
+
+#[actix_web::test]
+async fn should_authorize_gpay_payment() {
+    let authorize_response = Payu {}.authorize_payment(Some(types::PaymentsAuthorizeData{
+            payment_method_data: types::api::PaymentMethod::Wallet(api::WalletData{
+                    issuer_name: api_models::enums::WalletIssuer::GooglePay,
+                    token: Some("eyJzaWduYXR1cmUiOiJNRVFDSUEwYWN1RGF3SkxYbDZXSEhKMU5oWWJJdHU2cElnaUcwYjlmNHY2Q1ZpMlpBaUJKbE9SR0Z0ME5kVlp0T0h4QTJQa3NDY3ZFSERhVGt4eHFsVnNiTVRnc1dRXHUwMDNkXHUwMDNkIiwicHJvdG9jb2xWZXJzaW9uIjoiRUN2MSIsInNpZ25lZE1lc3NhZ2UiOiJ7XCJlbmNyeXB0ZWRNZXNzYWdlXCI6XCJyRkoxT1haOXNzQjdKTkxvV0pLVklLWGQ2RnZzeWwrUW5naUY3UWU1RlEvZWJMVXdWOGNUdnZmSnM4T0ptcEVWNGt5M2t5MCttNjhlanVXTlhrYm1lWmRmTFdVeEtFREkxVG5MMjYwVWtvZ1NNRDc1VEUyaVYwbFdDY2xKSnl0RXdmR0JmTWZYUVNPSGpUempOYTlTZmtyT21LTk0rTDRsNGlqNFNXWFZaUnlEVmZnajZ6TnNaV0hhbUZjZUZTLzFmOGRheHFSQzRTT2d5SEVjQ0ZrVEZ0RUFONk1HRlFVd2NOY2hRZml1TTliL3lqYmJKVXE1aEtZbXFPMXg1K0hxWE9wVHhkeWFSUTFDeFJoQWJZdi9ZU2xMaU5Ja09PZ1hnRjBkKytkTnhIcHBDTVVnbkRITytSQzZiaXoyZnFiRXFQWUgvVlRNNTFuRmRkRlcwVk1CWUxlcC9hTkRBak9OSUc4WjlJZ1c0ajhnTldBTWlUVm5xM3NjcDVvTDhyMHh5M1VtQnFYMnlPUCtaVHZneGdxYys0ZHhrTWhzWVVBcWpnUmpMa1BzNk1zZnhLaGUyODFpL0pmRlcxY2VSUW9uQkFcXHUwMDNkXFx1MDAzZFwiLFwiZXBoZW1lcmFsUHVibGljS2V5XCI6XCJCTUt0VnozQ3ZZYWNKOWVBN0pwWkVSUVVHMkIvaDFKZU1UQkdVc09wbERjcG50dVEwM0hjRXd3K1ZrRlBXVUlKZlJ3WnZyVjFOaVlGNm9iaWxobTBZNjhcXHUwMDNkXCIsXCJ0YWdcIjpcIlI3MWFVVGVrbzRGZVBibXhkekdtZVpDcS9VckVhK2dHd3VkT2RBUE9ZNEFcXHUwMDNkXCJ9In="
```

Comment:
Is this huge string literal for the token intended to be hard-coded in tests? Or is there a possibility that this can be calculated?